akka-cluster-typed/src/test/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionistSpec.scala

```diff
@@ -91,76 +89,241 @@ object ClusterReceptionistSpec {
+        clusterNode2.manager ! Join(clusterNode1.selfMember.address)
+        regProbe1.awaitAssert(clusterNode1.state.members.count(_.status == MemberStatus.Up))
+
+        // and the subscriber on node1 should see the service
+        val remoteServiceRefs = regProbe1.expectMessageType[Listing].serviceInstances(PingKey)
+        val theRef = remoteServiceRefs.head
+        theRef ! Ping(regProbe1.ref)
+        regProbe1.expectMessage(Pong)
+
+        // abrupt termination
+        system2.terminate()
+        regProbe1.expectMessage(10.seconds, Listing(PingKey, Set.empty[ActorRef[PingProtocol]]))
+      } finally {
+        testKit1.shutdownTestKit()
+        if (!system1.whenTerminated.isCompleted) testKit2.shutdownTestKit()
+      }
     }
 
-  }
+    "handle a new incarnation of the same node well" in {
+      val testKit1 = new ActorTestKit {
+        override def name = super.name + "-test-3"
+        override def config = ClusterReceptionistSpec.config
+      }
+      val system1 = testKit1.system
+      val testKit2 = new ActorTestKit {
+        override def name = system1.name
+        override def config = testKit1.system.settings.config
+      }
+      val system2 = testKit2.system
+      try {
+
+        val clusterNode1 = Cluster(system1)
+        clusterNode1.manager ! Join(clusterNode1.selfMember.address)
+        val clusterNode2 = Cluster(system2)
+        clusterNode2.manager ! Join(clusterNode1.selfMember.address)
+
+        val regProbe1 = TestProbe[Any]()(system1)
+        val regProbe2 = TestProbe[Any]()(system2)
+
+        regProbe1.awaitAssert(clusterNode1.state.members.count(_.status == MemberStatus.Up) == 2)
+
+        system1.receptionist ! Subscribe(PingKey, regProbe1.ref)
+        regProbe1.expectMessage(Listing(PingKey, Set.empty[ActorRef[PingProtocol]]))
+
+        val service2 = testKit2.spawn(pingPongBehavior, "instance")
+        system2.receptionist ! Register(PingKey, service2, regProbe2.ref)
+        regProbe2.expectMessage(Registered(PingKey, service2))
+
+        // make sure we saw the first incarnation on node1
+        val remoteServiceRefs = regProbe1.expectMessageType[Listing].serviceInstances(PingKey)
+        val theRef = remoteServiceRefs.head
+        theRef ! Ping(regProbe1.ref)
+        regProbe1.expectMessage(Pong)
+
+        // FIXME do we need to blackhole the connection to system2 before terminating
+        // right now it doesn't work anyways though ;D
+
+        // abrupt termination but then a node with the same host:port comes online quickly
+        system1.log.debug("Terminating system2, uid: [{}]", clusterNode2.selfMember.uniqueAddress.longUid)
+        Await.ready(system2.terminate(), 10.seconds)
+
+        val testKit3 = new ActorTestKit {
+          override protected def name: String = system1.name
+          def system1NodeString = {
+            val address = clusterNode1.selfMember.address
+            s"akka://${system1.name}@${address.host.get}:${address.port.get}"
```

Comment:
we are always using artery here, but we shouldn't hardcode the protocol name. wouldn't `clusterNode1.selfMember.address.toString` work?

Reply:
good catch

Reply:
Maybe we could just as well join programmatically also, I just wanted to trigger join as soon as possible on system start